test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -471,7 +471,23 @@ json_spirit::mObject fillBCTest(json_spirit::mObject const& _input, bool _allowI
     }
 
     output["blocks"] = blArray;
-    output["postState"] = fillJsonWithState(testChain.topBlock().state());
+
+    size_t totalCodeSize = 0;
+    bool isSmallState = testChain.topBlock().state().addresses().size() < 20;
+    if (isSmallState)
+    {
+        for (auto const& accAddr : testChain.topBlock().state().addresses())
+        {
+            totalCodeSize += testChain.topBlock().state().codeSize(accAddr.first);
+            if (totalCodeSize > 1024 * 1000)  // if totalCodeSize > 1Mb
+                isSmallState = false;
+        }
+    }
+
+    if (isSmallState || Options::get().fullstate)
```

Comment:
Check `fullstate` first, if it's true, it won't need to iterate over 20 addresses